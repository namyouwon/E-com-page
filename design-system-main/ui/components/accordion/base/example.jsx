// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import uniqueId from 'lodash.uniqueid';
import Accordion, { AccordionSection } from '../';
import { ActionOverflow } from '../../../components/menus/action-overflow/example';
import Card from '../../cards/';

export const AccordionExample = () => (
  <Accordion>
    <AccordionSection
      summary="Accordion summary"
      isOpen
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - A
    </AccordionSection>
    <AccordionSection
      summary="Accordion summary"
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - B
    </AccordionSection>
    <AccordionSection
      summary="Accordion summary"
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - C
    </AccordionSection>
  </Accordion>
);

export default <AccordionExample />;

export let states = [
  {
    id: 'section-one-open',
    label: 'Section one open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-two-open',
    label: 'Section two open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-three-open',
    label: 'Section three open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'multiple-sections-open',
    label: 'Multiple sections open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  }
];

export let examples = [
  {
    id: 'long-heading-truncated',
    label: 'Long Section Heading with Truncation for larger screens',
    demoStyles: 'max-width: 500px;',
    storybookStyles: true,
    element: (
      <Card>
        <Accordion>
          <AccordionSection
            summary="Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary"
            isOpen
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - A
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - B
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - C
          </AccordionSection>
        </Accordion>
      </Card>
    )
  },
  {
    id: 'nested-accordions',
    label: 'Nested Accordions',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          <Accordion>
            <AccordionSection
              summary="Nested accordion summary"
              isOpen
              referenceId={uniqueId('referenceId-')}
            >
              Accordion details - A
            </AccordionSection>
            <AccordionSection
              summary="Nested accordion summary"
              referenceId={uniqueId('referenceId-')}
            >
              Accordion details - B
            </AccordionSection>
          </Accordion>
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'styled',
    label: 'Wrapped in Card',
    element: (
      <div className="slds-card">
        <Accordion>
          <AccordionSection
            summary="Accordion summary"
            isOpen
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - A
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - B
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - C
          </AccordionSection>
        </Accordion>
      </div>
    )
  },
  {
    id: 'has-action-overflow',
    label: 'Has Action Overflow Menu',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-background-color',
    label: 'Accordion Section - Background Color',
    demoStyles: '--slds-c-accordion-summary-color-background: lightblue',
    storybookStyles: true,
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-background-color',
    label: 'Accordion Section - Text Color',
    demoStyles:
      '--slds-c-accordion-heading-text-color: green; --slds-c-accordion-heading-text-color-hover: red',
    storybookStyles: true,
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  }
];
