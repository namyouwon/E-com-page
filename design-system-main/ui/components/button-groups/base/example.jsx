// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { Button } from '../../buttons/base/example';
import ButtonIcon from '../../button-icons/';
import {
  Menu,
  MenuList,
  MenuItem,
  Trigger
} from '../../menus/dropdown/example';
import classNames from 'classnames';

/// ////////////////////////////////////////
// Partial(s)
/// ////////////////////////////////////////

export let ButtonGroup = props => (
  <div
    className={classNames('slds-button-group', props.className)}
    role="group"
  >
    {props.children}
  </div>
);

export let ButtonGroupList = props => (
  <ul className={classNames('slds-button-group-list', props.className)}>
    {props.children}
  </ul>
);

/// ////////////////////////////////////////
// Export
/// ////////////////////////////////////////

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <ButtonGroup>
        <Button isNeutral>Refresh</Button>
        <Button isNeutral>Edit</Button>
        <Button isNeutral>Save</Button>
      </ButtonGroup>
    )
  }
];

export let states = [
  {
    id: 'disabled',
    label: 'Disabled',
    element: (
      <ButtonGroup>
        <Button isNeutral>Refresh</Button>
        <Button isNeutral disabled>
          Edit
        </Button>
        <Button isNeutral>Save</Button>
      </ButtonGroup>
    )
  },
  {
    id: 'with-overflow',
    label: 'With Overflow Menu',
    element: (
      <ButtonGroup>
        <Button isNeutral>Refresh</Button>
        <Button isNeutral>Edit</Button>
        <Button isNeutral>Save</Button>
        <Trigger className="slds-button_last" />
      </ButtonGroup>
    )
  },
  {
    id: 'with-overflow-open',
    label: 'With Overflow Menu - Open',
    element: (
      <div className="demo-only" style={{ height: '8.75rem' }}>
        <ButtonGroup>
          <Button isNeutral>Refresh</Button>
          <Button isNeutral>Edit</Button>
          <Button isNeutral>Save</Button>
          <Trigger className="slds-button_last" isOpen>
            <Menu className="slds-dropdown_right slds-dropdown_actions">
              <MenuList>
                <MenuItem tabIndex="0">Overflow Item One</MenuItem>
                <MenuItem>Overflow Item Two</MenuItem>
                <MenuItem>Overflow Item Three</MenuItem>
              </MenuList>
            </Menu>
          </Trigger>
        </ButtonGroup>
      </div>
    )
  },
  {
    id: 'with-overflow-disabled',
    label: 'With Overflow Menu - Disabled',
    element: (
      <ButtonGroup>
        <Button isNeutral>Refresh</Button>
        <Button isNeutral>Edit</Button>
        <Button isNeutral>Save</Button>
        <div className="slds-dropdown-trigger slds-dropdown-trigger_click slds-button_last">
          <ButtonIcon
            aria-haspopup="true"
            assistiveText="More Actions"
            className="slds-button_icon-border-filled"
            disabled
            symbol="down"
            title="More Actions"
          />
        </div>
      </ButtonGroup>
    )
  }
];

export let examples = [
  {
    id: 'inverse',
    label: 'Inverse',
    element: (
      <div
        className="demo-only"
        style={{ padding: '0.5rem', background: '#16325c' }}
      >
        <ButtonGroup>
          <Button isInverse>Refresh</Button>
          <Button isInverse>Edit</Button>
          <Button isInverse>Save</Button>
        </ButtonGroup>
      </div>
    )
  },
  {
    id: 'inverse-disabled',
    label: 'Inverse - Disabled',
    element: (
      <div
        className="demo-only"
        style={{ padding: '0.5rem', background: '#16325c' }}
      >
        <ButtonGroup>
          <Button isInverse>Refresh</Button>
          <Button isInverse disabled>
            Edit
          </Button>
          <Button isInverse>Save</Button>
        </ButtonGroup>
      </div>
    )
  },
  {
    id: 'inverse-overflow',
    label: 'Inverse - With Overflow Menu',
    element: (
      <div
        className="demo-only"
        style={{ padding: '0.5rem', background: '#16325c' }}
      >
        <ButtonGroup>
          <Button isInverse>Refresh</Button>
          <Button isInverse>Edit</Button>
          <Button isInverse>Save</Button>
          <div className="slds-dropdown-trigger slds-dropdown-trigger_click slds-button_last">
            <ButtonIcon
              className="slds-button_icon-border-inverse"
              assistiveText="More Actions"
              aria-haspopup="true"
              symbol="down"
              title="More Actions"
            />
          </div>
        </ButtonGroup>
      </div>
    )
  },
  {
    id: 'inverse-overflow-disabled',
    label: 'Inverse - With Overflow Menu - Disabled',
    element: (
      <div
        className="demo-only"
        style={{ padding: '0.5rem', background: '#16325c' }}
      >
        <ButtonGroup>
          <Button isInverse>Refresh</Button>
          <Button isInverse>Edit</Button>
          <Button isInverse>Save</Button>
          <div className="slds-dropdown-trigger slds-dropdown-trigger_click slds-button_last">
            <ButtonIcon
              className="slds-button_icon-border-inverse"
              assistiveText="More Actions"
              disabled
              aria-haspopup="true"
              symbol="down"
              title="More Actions"
            />
          </div>
        </ButtonGroup>
      </div>
    )
  },
  {
    id: 'button-group-icon',
    label: 'Button Icon Group',
    element: (
      <ButtonGroup>
        <ButtonIcon
          className="slds-button_icon-border-filled"
          symbol="chart"
          assistiveText="Charts"
          title="Charts"
          aria-pressed="false"
        />
        <ButtonIcon
          className="slds-button_icon-border-filled"
          symbol="filterList"
          assistiveText="Filter List"
          title="Filter List"
          aria-pressed="false"
        />
        <ButtonIcon
          className="slds-button_icon-more"
          hasDropdown
          assistiveText="More Actions"
          title="More Actions"
        />
      </ButtonGroup>
    )
  },
  {
    id: 'button-group-icon-disabled',
    label: 'Button Icon Group - Disabled',
    element: (
      <ButtonGroup>
        <ButtonIcon
          className="slds-button_icon-border-filled"
          symbol="chart"
          assistiveText="Charts"
          title="Charts"
          aria-pressed="false"
        />
        <ButtonIcon
          className="slds-button_icon-border-filled"
          disabled
          symbol="filterList"
          assistiveText="Filter List"
          title="Filter List"
          aria-pressed="false"
        />
        <ButtonIcon
          className="slds-button_icon-more"
          hasDropdown
          assistiveText="More Actions"
          title="More Actions"
        />
      </ButtonGroup>
    )
  },
  {
    id: 'button-group-icon-selected',
    label: 'Button Icon Group - Selected',
    element: (
      <ButtonGroup>
        <ButtonIcon
          className="slds-button_icon-border-filled slds-is-selected"
          symbol="chart"
          assistiveText="Charts"
          title="Charts"
          aria-pressed="true"
        />
        <ButtonIcon
          className="slds-button_icon-border-filled"
          symbol="filterList"
          assistiveText="Filter List"
          title="Filter List"
          aria-pressed="false"
        />
        <ButtonIcon
          className="slds-button_icon-more"
          hasDropdown
          assistiveText="More Actions"
          title="More Actions"
        />
      </ButtonGroup>
    )
  },
  {
    id: 'brand-with-overflow',
    label: 'Brand Button With Overflow Menu',
    element: (
      <ButtonGroup>
        <Button isBrand>Save</Button>
        <div className="slds-dropdown-trigger slds-dropdown-trigger_click slds-button_last">
          <ButtonIcon
            aria-haspopup="true"
            assistiveText="More Actions"
            className="slds-button_icon-brand"
            symbol="down"
            title="More Actions"
          />
        </div>
      </ButtonGroup>
    )
  },
  {
    id: 'style-encapsulated-button-group',
    label: 'Button Group with Style Encapsulation',
    element: (
      <ButtonGroup>
        <div>
          <Button isNeutral isFirst>
            First
          </Button>
        </div>
        <div>
          <Button isNeutral isMiddle>
            Middle
          </Button>
        </div>
        <div>
          <Button isNeutral isMiddle>
            Middle
          </Button>
        </div>
        <div>
          <Button isNeutral isLast>
            Last
          </Button>
        </div>
      </ButtonGroup>
    )
  },
  {
    id: 'style-encapsulated-button-group-2-button',
    label: 'Button Group with Style Encapsulation - First/Last',
    element: (
      <ButtonGroup>
        <div>
          <Button isNeutral isFirst>
            First
          </Button>
        </div>
        <div>
          <Button isNeutral isLast>
            Last
          </Button>
        </div>
      </ButtonGroup>
    )
  },
  {
    id: 'style-encapsulated-button-icon-group',
    label: 'Button Icon Group with Style Encapsulation',
    element: (
      <ButtonGroup>
        <div>
          <ButtonIcon
            className="slds-button_icon-border-filled"
            symbol="chart"
            assistiveText="Charts"
            title="Charts"
            aria-pressed="false"
            isFirst
          />
        </div>
        <div>
          <ButtonIcon
            className="slds-button_icon-border-filled"
            symbol="filterList"
            assistiveText="Filter List"
            title="Filter List"
            aria-pressed="false"
            isMiddle
          />
        </div>
        <div>
          <ButtonIcon
            className="slds-button_icon-border-filled"
            symbol="comments"
            assistiveText="Collaborate"
            title="Collaborate"
            aria-pressed="false"
            isMiddle
          />
        </div>
        <div>
          <ButtonIcon
            className="slds-button_icon-more"
            hasDropdown
            assistiveText="More Actions"
            title="More Actions"
            isLast
          />
        </div>
      </ButtonGroup>
    )
  },
  {
    id: 'style-encapsulated-button-icon-group-2-button',
    label: 'Button Icon Group with Style Encapsulation - First/Last',
    element: (
      <ButtonGroup>
        <div>
          <ButtonIcon
            className="slds-button_icon-border-filled"
            symbol="chart"
            assistiveText="Charts"
            title="Charts"
            aria-pressed="false"
            isFirst
          />
        </div>
        <div>
          <ButtonIcon
            className="slds-button_icon-more"
            hasDropdown
            assistiveText="More Actions"
            title="More Actions"
            isLast
          />
        </div>
      </ButtonGroup>
    )
  }
];
