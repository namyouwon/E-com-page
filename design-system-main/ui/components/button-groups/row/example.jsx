// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '../../buttons/base/example';
import ButtonIcon from '../../button-icons/';
import { ButtonGroup } from '../base/example';

export const ButtonGroupRow = props => (
  <ul className="slds-button-group-row">{props.children}</ul>
);

ButtonGroupRow.propTypes = {
  children: PropTypes.node
};

export const ButtonGroupItem = props => (
  <li className="slds-button-group-item">{props.children}</li>
);

ButtonGroupItem.propTypes = {
  children: PropTypes.node
};

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <ButtonGroupRow>
        <ButtonGroupItem>
          <Button isNeutral>Refresh</Button>
        </ButtonGroupItem>
        <ButtonGroupItem>
          <Button isNeutral>Edit</Button>
        </ButtonGroupItem>
        <ButtonGroupItem>
          <Button isBrand>Save</Button>
        </ButtonGroupItem>
      </ButtonGroupRow>
    )
  }
];

export let examples = [
  {
    id: 'buttons-icons',
    label: 'Row of button icons',
    element: (
      <ButtonGroupRow>
        <ButtonGroupItem>
          <div className="slds-dropdown-trigger slds-dropdown-trigger_clicked">
            <ButtonIcon
              hasDropdown
              theme="neutral"
              assistiveText="More options"
              title="More Options"
            />
          </div>
        </ButtonGroupItem>
        <ButtonGroupItem>
          <div className="slds-dropdown-trigger slds-dropdown-trigger_clicked">
            <ButtonIcon
              hasDropdown
              theme="neutral"
              symbol="table"
              assistiveText="More options"
              title="More Options"
            />
          </div>
        </ButtonGroupItem>
        <ButtonGroupItem>
          <ButtonIcon symbol="edit" theme="neutral" />
        </ButtonGroupItem>
        <ButtonGroupItem>
          <ButtonIcon symbol="refresh" theme="neutral" />
        </ButtonGroupItem>
        <ButtonGroupItem>
          <ButtonGroup>
            <ButtonIcon symbol="chart" theme="neutral" />
            <ButtonIcon symbol="filter" theme="neutral" />
          </ButtonGroup>
        </ButtonGroupItem>
      </ButtonGroupRow>
    )
  }
];
