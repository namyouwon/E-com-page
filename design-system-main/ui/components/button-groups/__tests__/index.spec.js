// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license
/* eslint-env jest */

import React from 'react';
import createHelpers from '../../../../jest.helpers';
import { Button } from '../../buttons/base/example';
import ButtonIcon from '../../button-icons/';
import { ButtonGroup } from '../base/example';
import { ButtonGroupRow, ButtonGroupItem } from '../row/example';
import {
  Menu,
  MenuList,
  MenuItem,
  Trigger
} from '../../menus/dropdown/example';

const { matchesMarkup } = createHelpers(__dirname);

it('renders a default button group', () =>
  matchesMarkup(
    <ButtonGroup>
      <Button isNeutral>Refresh</Button>
      <Button isNeutral>Edit</Button>
      <Button isNeutral>Save</Button>
    </ButtonGroup>
  ));

it('renders a button group with overflow open', () =>
  matchesMarkup(
    <div className="demo-only" style={{ height: '8.75rem' }}>
      <ButtonGroup>
        <Button isNeutral>Refresh</Button>
        <Button isNeutral>Edit</Button>
        <Button isNeutral>Save</Button>
        <Trigger className="slds-button_last" isOpen>
          <Menu className="slds-dropdown_right slds-dropdown_actions">
            <MenuList>
              <MenuItem tabIndex="0">Overflow Item One</MenuItem>
              <MenuItem>Overflow Item Two</MenuItem>
              <MenuItem>Overflow Item Three</MenuItem>
            </MenuList>
          </Menu>
        </Trigger>
      </ButtonGroup>
    </div>
  ));

it('renders a row a buttons spaced out', () =>
  matchesMarkup(
    <ButtonGroupRow>
      <ButtonGroupItem>
        <Button isNeutral>Refresh</Button>
      </ButtonGroupItem>
      <ButtonGroupItem>
        <Button isNeutral>Edit</Button>
      </ButtonGroupItem>
      <ButtonGroupItem>
        <Button isBrand>Save</Button>
      </ButtonGroupItem>
    </ButtonGroupRow>
  ));

it('renders a row a button icons spaced out', () =>
  matchesMarkup(
    <ButtonGroupRow>
      <ButtonGroupItem>
        <div className="slds-dropdown-trigger slds-dropdown-trigger_clicked">
          <ButtonIcon
            hasDropdown
            theme="neutral"
            assistiveText="More options"
            title="More Options"
          />
        </div>
      </ButtonGroupItem>
      <ButtonGroupItem>
        <div className="slds-dropdown-trigger slds-dropdown-trigger_clicked">
          <ButtonIcon
            hasDropdown
            theme="neutral"
            symbol="table"
            assistiveText="More options"
            title="More Options"
          />
        </div>
      </ButtonGroupItem>
      <ButtonGroupItem>
        <ButtonIcon symbol="edit" theme="neutral" />
      </ButtonGroupItem>
      <ButtonGroupItem>
        <ButtonIcon symbol="refresh" theme="neutral" />
      </ButtonGroupItem>
      <ButtonGroupItem>
        <ButtonGroup>
          <ButtonIcon symbol="chart" theme="neutral" />
          <ButtonIcon symbol="filter" theme="neutral" />
        </ButtonGroup>
      </ButtonGroupItem>
    </ButtonGroupRow>
  ));

it('renders a single brand button with brand button icon dropdown', () =>
  matchesMarkup(
    <ButtonGroup>
      <Button isBrand>Save</Button>
      <div className="slds-dropdown-trigger slds-dropdown-trigger_click slds-button_last">
        <ButtonIcon
          aria-haspopup="true"
          assistiveText="More Actions"
          className="slds-button_icon-brand"
          symbol="down"
          title="More Actions"
        />
      </div>
    </ButtonGroup>
  ));

describe('Style Encapsulation Markup & Classes', () => {
  it('renders a encapsulated button with the `slds-button_first` class', () =>
    matchesMarkup(
      <ButtonGroup>
        <div>
          <Button isNeutral isFirst>
            First
          </Button>
        </div>
      </ButtonGroup>
    ));

  it('renders a encapsulated button with the `slds-button_middle` class', () =>
    matchesMarkup(
      <ButtonGroup>
        <div>
          <Button isNeutral isMiddle>
            Middle
          </Button>
        </div>
      </ButtonGroup>
    ));

  it('renders a encapsulated button with the `slds-button_last` class', () =>
    matchesMarkup(
      <ButtonGroup>
        <div>
          <Button isNeutral isFirst>
            Last
          </Button>
        </div>
      </ButtonGroup>
    ));
});
