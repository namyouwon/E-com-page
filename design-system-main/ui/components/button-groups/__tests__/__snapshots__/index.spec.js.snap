// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Style Encapsulation Markup & Classes renders a encapsulated button with the \`slds-button_first\` class 1`] = `
<div
  className="slds-button-group"
  role="group"
>
  <div>
    <button
      className="slds-button slds-button_neutral slds-button_first"
      onClick={[Function]}
    >
      First
    </button>
  </div>
</div>
`;

exports[`Style Encapsulation Markup & Classes renders a encapsulated button with the \`slds-button_last\` class 1`] = `
<div
  className="slds-button-group"
  role="group"
>
  <div>
    <button
      className="slds-button slds-button_neutral slds-button_first"
      onClick={[Function]}
    >
      Last
    </button>
  </div>
</div>
`;

exports[`Style Encapsulation Markup & Classes renders a encapsulated button with the \`slds-button_middle\` class 1`] = `
<div
  className="slds-button-group"
  role="group"
>
  <div>
    <button
      className="slds-button slds-button_neutral slds-button_middle"
      onClick={[Function]}
    >
      Middle
    </button>
  </div>
</div>
`;

exports[`renders a button group with overflow open 1`] = `
<div
  className="demo-only"
  style={
    Object {
      "height": "8.75rem",
    }
  }
>
  <div
    className="slds-button-group"
    role="group"
  >
    <button
      className="slds-button slds-button_neutral"
      onClick={[Function]}
    >
      Refresh
    </button>
    <button
      className="slds-button slds-button_neutral"
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      className="slds-button slds-button_neutral"
      onClick={[Function]}
    >
      Save
    </button>
    <div
      className="slds-dropdown-trigger slds-dropdown-trigger_click slds-is-open slds-button_last"
    >
      <button
        aria-expanded="true"
        aria-haspopup="true"
        className="slds-button slds-button_icon slds-button_icon-border-filled"
        onClick={[Function]}
        title="Show More"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Show More
        </span>
      </button>
      <div
        className="slds-dropdown slds-dropdown_right slds-dropdown_actions"
      >
        <ul
          className="slds-dropdown__list"
          role="menu"
        >
          <li
            className="slds-dropdown__item"
            role="presentation"
          >
            <a
              href="#"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
            >
              <span
                className="slds-truncate"
                title="Overflow Item One"
              >
                Overflow Item One
              </span>
            </a>
          </li>
          <li
            className="slds-dropdown__item"
            role="presentation"
          >
            <a
              href="#"
              onClick={[Function]}
              role="menuitem"
              tabIndex="-1"
            >
              <span
                className="slds-truncate"
                title="Overflow Item Two"
              >
                Overflow Item Two
              </span>
            </a>
          </li>
          <li
            className="slds-dropdown__item"
            role="presentation"
          >
            <a
              href="#"
              onClick={[Function]}
              role="menuitem"
              tabIndex="-1"
            >
              <span
                className="slds-truncate"
                title="Overflow Item Three"
              >
                Overflow Item Three
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders a default button group 1`] = `
<div
  className="slds-button-group"
  role="group"
>
  <button
    className="slds-button slds-button_neutral"
    onClick={[Function]}
  >
    Refresh
  </button>
  <button
    className="slds-button slds-button_neutral"
    onClick={[Function]}
  >
    Edit
  </button>
  <button
    className="slds-button slds-button_neutral"
    onClick={[Function]}
  >
    Save
  </button>
</div>
`;

exports[`renders a row a button icons spaced out 1`] = `
<ul
  className="slds-button-group-row"
>
  <li
    className="slds-button-group-item"
  >
    <div
      className="slds-dropdown-trigger slds-dropdown-trigger_clicked"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        className="slds-button slds-button_icon slds-button_icon-more"
        onClick={[Function]}
        title="More Options"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_x-small"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          More options
        </span>
      </button>
    </div>
  </li>
  <li
    className="slds-button-group-item"
  >
    <div
      className="slds-dropdown-trigger slds-dropdown-trigger_clicked"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        className="slds-button slds-button_icon slds-button_icon-more"
        onClick={[Function]}
        title="More Options"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#table"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_x-small"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          More options
        </span>
      </button>
    </div>
  </li>
  <li
    className="slds-button-group-item"
  >
    <button
      className="slds-button slds-button_icon slds-button_icon-border-filled"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#edit"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Provide description of action
      </span>
    </button>
  </li>
  <li
    className="slds-button-group-item"
  >
    <button
      className="slds-button slds-button_icon slds-button_icon-border-filled"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#refresh"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Provide description of action
      </span>
    </button>
  </li>
  <li
    className="slds-button-group-item"
  >
    <div
      className="slds-button-group"
      role="group"
    >
      <button
        className="slds-button slds-button_icon slds-button_icon-border-filled"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chart"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Provide description of action
        </span>
      </button>
      <button
        className="slds-button slds-button_icon slds-button_icon-border-filled"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#filter"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Provide description of action
        </span>
      </button>
    </div>
  </li>
</ul>
`;

exports[`renders a row a buttons spaced out 1`] = `
<ul
  className="slds-button-group-row"
>
  <li
    className="slds-button-group-item"
  >
    <button
      className="slds-button slds-button_neutral"
      onClick={[Function]}
    >
      Refresh
    </button>
  </li>
  <li
    className="slds-button-group-item"
  >
    <button
      className="slds-button slds-button_neutral"
      onClick={[Function]}
    >
      Edit
    </button>
  </li>
  <li
    className="slds-button-group-item"
  >
    <button
      className="slds-button slds-button_brand"
      onClick={[Function]}
    >
      Save
    </button>
  </li>
</ul>
`;

exports[`renders a single brand button with brand button icon dropdown 1`] = `
<div
  className="slds-button-group"
  role="group"
>
  <button
    className="slds-button slds-button_brand"
    onClick={[Function]}
  >
    Save
  </button>
  <div
    className="slds-dropdown-trigger slds-dropdown-trigger_click slds-button_last"
  >
    <button
      aria-haspopup="true"
      className="slds-button slds-button_icon slds-button_icon-brand"
      onClick={[Function]}
      title="More Actions"
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        More Actions
      </span>
    </button>
  </div>
</div>
`;
